joblib/test/test_hashing.py

```diff
@@ -259,6 +259,14 @@ def test_hash_object_dtype():
     nose.tools.assert_equal(hash(a),
                             hash(b))
 
+    from ..parallel import Parallel, delayed
+
+    args = {'np1': np.arange(10), 'np2': np.arange(20)}
+    codes = Parallel(n_jobs=3)(delayed(hash)(args) for _ in range(10))
+
+    nose.tools.assert_equal(len(set(codes)), 1)
+    nose.tools.assert_equal(codes[0], hash(args))
```

Comment:
The test does fail in my machine ( windows 8 & fedora 20, Python 2 & 3). I don't know why it succeed in your environment .

When dump python object, the pickler save the id of object's attribute to memo to allow shared or recursive objects to pickled. That is if the current attribute's id is in memo, there is only a reference to be saved. 

in the normal case, the dtype attribute of two ndarray id is identical, meaning that id(a.dtype) == id(b.dtype). However, when the ndarrays is loaded using pickle, this equality is ruined. So, in this situation, the second dtype object isn't  dumped as a reference.

I  thought that if two dtype objects are equal(though they may not identical), then the id of repr of them are equal. However, I did not find any doc to prove it. Maybe, hash(repr(obj.dtype)) is better since pickler does not memorize int.